### PR TITLE
add api for user to create org

### LINKED ISSUE #5260: API endpoint for organization creation
- Gitea version (or commit ref): 1.5.3
- Git version: not relevant (website issue)
- Operating system: Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
It seems that there is no API endpoint to create an organization (at least I can not find it).

I expected something like ``POST /orgs/`` but this does not exist in swagger:
https://try.gitea.io/api/swagger#/organization
The function is also missing in the go client library of Gitea: https://github.com/go-gitea/go-sdk/blob/master/gitea/org.go

I also found an old pull request for gogs: gogs/gogs#4835
There an API endpoint for the creation of organizations was added.

### PR DESCRIPTION
**Pull request by lunny**
This will fix #5260. And github API v3 doesn't implement this route.

### CODE REVIEW

**Comment by zeripath** [comment]
@lunny this appears to be failing to build because there's a newline at the end of the committed `templates/swagger/v1_json.tmpl` - I think if you just run: `make swagger-generate` and commit that it should fix it.

**Review by appleboy — APPROVED** [review]

**Comment by lunny** [comment]
@zeripath the newline at the end of v1_json.tmpl is generated by `make swagger-generate`. I have fixed it.

**Comment by sapk** [comment]
@lunny CI failed (401 Unauthorized)

**Comment by lunny** [comment]
@sapk fixed.

**Review by zeripath — APPROVED** [review]

File: routers/api/v1/org/org.go [hunk 1]
```diff
@@ -62,6 +62,56 @@ func ListUserOrgs(ctx *context.APIContext) {
 	listUserOrgs(ctx, u, false)
 }
 
+// Create api for create organization
```
Comment (techknowlogick) [inline routers/api/v1/org/org.go:66]: Can you add Gitea copyright to header of this file?

**Review by techknowlogick — CHANGES_REQUESTED** [review]

File: routers/api/v1/api.go [hunk 2]
```diff
@@ -578,6 +578,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 		// Organizations
 		m.Get("/user/orgs", reqToken(), org.ListMyOrgs)
 		m.Get("/users/:username/orgs", org.ListUserOrgs)
+		m.Post("/orgs", reqToken(), bind(api.CreateOrgOption{}), org.Create)
```
Comment (techknowlogick) [inline routers/api/v1/api.go:582]: Can you add Gitea copyright to header of this file?

File: routers/api/v1/org/org.go (hunk 1 above)
Comment (zeripath) [reply routers/api/v1/org/org.go:66]: Ugh sorry @techknowlogick forgot to look for that!

Comment (techknowlogick) [reply routers/api/v1/org/org.go:66]: @zeripath it's ok. It's something that is missed a lot, and we thankfully have git so if we do miss in one PR it we can just look at the history of the repo and add the appropriate dates when fixing.

**Comment by lunny** [comment]
@techknowlogick done.

**Review by lafriks — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by lunny** [comment]
@techknowlogick need your approval.

**Review by techknowlogick — APPROVED** [review]
@lunny thanks for reminder 😄 LGTM

**Pull request merged**